packages/backend/src/peripherals/database/BlockTransactionCountRepository.ts

```diff
@@ -50,15 +57,52 @@ export class BlockTransactionCountRepository extends BaseRepository {
     return rows.length
   }
 
+  async refreshProjectTip(projectId: ProjectId) {
+    const knex = await this.knex()
+    const currentTip = await this.getTipByProject(projectId)
+    const tipNumber =
+      (await this.getFirstBlockNumberWithoutNext(
+        projectId,
+        currentTip?.block_number,
+      )) ?? (await this.getMaxBlockNumber(projectId))
```

Comment:
What is the edge case for this.getMaxBlockNumber? When will this.getFirstBlockNumberWithoutNext return undefined but this.getMaxBlockNumber not?

Reply:
`getFirstBlockNumberWithoutNext` returns first row without next. If there are no gaps then there will be no such row, because `max` is the tip and hence we return `max`.